recipes/fontconfig/meson/conanfile.py

```diff
@@ -79,12 +82,23 @@ def generate(self):
         })
         tc.generate()
 
+    def _replace_configuration_string(self, file, variable, old_value, new_value):
+        replace_in_file(self, file, f"conf.set_quoted('{variable}', {old_value})", f"conf.set_quoted('{variable}', {new_value})")
+
     def _patch_files(self):
         apply_conandata_patches(self)
-        replace_in_file(self, os.path.join(self.source_folder, "meson.build"),
+
+        build_script = os.path.join(self.source_folder, "meson.build")
+        replace_in_file(self, build_script,
                         "freetype_req = '>= 21.0.15'",
                         f"freetype_req = '{Version(self.dependencies['freetype'].ref.version)}'")
 
+        if self.settings.os in ("Linux", "FreeBSD") and self.options.get_safe("system_dirs", False):
+            self._replace_configuration_string(build_script, "CONFIGDIR", "fc_configdir", "'/etc/fonts/conf.d'")
+            self._replace_configuration_string(build_script, "FC_CACHEDIR", "fc_cachedir", "'/var/cache/fontconfig'")
+            self._replace_configuration_string(build_script, "FC_TEMPLATEDIR", "fc_templatedir", "'/usr/share/fontconfig/conf.avail'")
+            self._replace_configuration_string(build_script, "FONTCONFIG_PATH", "fc_baseconfigdir", "'/etc/fonts'")
```

Comment:
You should be able to accomplish the same without patching simply via project_options in generate(), I think:
```
tc.project_options["fc_configdir"] = "/etc/fonts/conf.d"
tc.project_options["fc_cachedir"] = "/var/cache/fontconfig"
tc.project_options["fc_templatedir"] = "/usr/share/fontconfig/conf.avail"
tc.project_options["fc_baseconfigdir"] = "/etc/fonts"
```
Also, you should probably override `xml-dir` as well? https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/main/meson_options.txt?ref_type=heads#L32-45

Reply:
I thought I had tried that previously, but you are right, it works on Conan 2.

Unfortunately, on Conan 1, Meson attempts to install something to /usr and fails. Looks like it is due to missing this change: https://github.com/conan-io/conan/pull/15706